calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -95,19 +96,19 @@ def _configure_interface(self):
         return ep
 
     def _get_node_ip(self):
-        """Determine the IP for the host node.
-
-        This hits a well-known IP (the Google public DNS).
+        """ 
+            Determine the IP for the host node.
         """
-        # We'd like to use the k8s API here, but it doesn't work in the Vagrant
-        # case since the node's name is set to its IP (so we have no way of
-        # getting from hostname=>IP).
-        # TODO: do this more reliably by parsing 'ip
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        s.connect(('8.8.8.8', 80))
-        ip = s.getsockname()[0]
-        s.close()
-        return ip
+        # Compile list of IPv4 addresses on network, return the first entry
+        ipv4_addresses = get_host_ips(version=4)
+        
+        # If no IPv4 addresses exist, check IPv6, return the first entry
+        if not ipv4_addresses:
+            return get_host_ips(version=6)[0]
```

Comment:
Right - I missed where this code was.

Returning None is probably not the right thing to do, since the calling code is just going to fall over.
Probably best to include a good error log and sys.exit(1)